api/src/main/java/org/apache/iceberg/io/FileIO.java

```diff
@@ -42,6 +46,30 @@ default InputFile newInputFile(String path, long length) {
     return newInputFile(path);
   }
 
+  default InputFile newInputFile(DataFile file) {
+    Preconditions.checkArgument(
+        file.keyMetadata() == null,
+        "Cannot decrypt data file: {} (use EncryptingFileIO)",
```

Comment:
nit: Suggestion on error

 Cannot decrypt encrypted data file without configured EncryptingFileIO: {}

Reply:
I think it's clear that the data file is encrypted, so I would omit that addition. And the "without configured EncryptingFileIO" is a stronger statement that isn't quite right. You can decrypt a data file using an `EncryptionManager` and `newInputFile(String)` but if you want to use this method, you must use `EncryptingFileIO`.